src/api/Snapshot.ts

```diff
@@ -212,16 +205,56 @@ export class Snapshot extends API {
 
     let votes: SnapshotVote[] = []
     while (hasNext) {
-      const result = await this.fetch<SnapshotVoteResponse>(`/graphql`, this.options()
-        .method('POST')
-        .json({ query, variables: { space, proposal, skip, first } })
+      const result = await this.fetch<SnapshotVoteResponse>(
+        `/graphql`,
+        this.options().method('POST').json({ query, variables: { space, proposal, skip, first } })
+      )
+
+      const currentVotes = result?.data?.votes || []
+      votes = [...votes, ...currentVotes]
+
+      if (currentVotes.length < first) {
+        hasNext = false
+      } else {
+        skip = currentVotes.length
+      }
+    }
+
+    return votes
+  }
+
+  async getAddressVotes(space: string, address: string) {
+    let hasNext = true
+    let skip = 0
+    const first = 500
+    const query = `
+      query Votes($space: String!, $address: String!, $first: Int!, $skip: Int!) {
+        votes (
+          first: $first,
+          skip: $skip,
+          where: {
+            space: $space,
+            voter: $address
+          },
+          orderBy: "created",
+          orderDirection: asc
+        ) {
+          id
+          voter
+          created
+        }
+      }
+    `
+
+    let votes: SnapshotVote[] = []
```

Comment:
all this part is the same as the one in `getProposalVotes`, but instead of sending the proposal we are using the address. Maybe we can extract this to a fn?

Reply:
Agree! Maybe we can talk later and see how can we extract it.